_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -1027,6 +1038,23 @@ public static function jetpack_connection_test() {
 		}
 	}
 
+	/**
+	 * Get plugins connected to the Jetpack.
+	 *
+	 * @return array|WP_Error WP_Error returned if connection test does not succeed.
+	 */
+	public static function get_connection_plugins() {
+		$plugins = ( new Connection_Manager() )->get_connected_plugins();
```

Comment:
My plan is to encapsulate us much as we can from the public usage.

We have `Connection_Manager` providing the main API for other plugins to use.
We also have the `Plugin` which reflects an instance of a particular plugin. Other plugins don't necessarily need to use it directly, but they can.
These two classes need to be backward compatible.

The `Plugin_Storage` is an internal class, and although developers can still use it in their plugins, we don't put it in the docs, so compatibility with older versions shouldn't be such a huge concern for us.
We would be able to get rid of some proxy methods at the cost of complicating the package maintenance, and making the package public interface more complex and harder to use.

Either way, it's only an opinion, and I'd live to hear your thoughts on this.

Reply:
That sounds fine. (I think we'll still need to be careful with backward compatibility for the `Plugins_Storage` class or any class with public methods just in case they're used externally.)

Reply:
You're right, we still need to take that into account.
We can't simply mark a class as `internal` for the package, and even if we could, I don't know if it would be a good idea after all.